src/js/common/schemaform/reducers/index.js

```diff
@@ -17,16 +17,30 @@ import { SET_DATA,
 } from '../actions';
 
 function recalculateSchemaAndData(initialState) {
-  return Object.keys(_.omit(['privacyAgreementAccepted', 'submission'], initialState))
+  const retVal = Object.keys(_.omit(['privacyAgreementAccepted', 'submission'], initialState))
     .reduce((state, pageKey) => {
       const page = state[pageKey];
       // on each data change, we need to do the following steps
+
+      // Flatten the data from all the pages
+      const formData = Object.keys(state).reduce((carry, pageName) => {
```

Comment:
I wouldn't worry too much about it. I think you need to remove the `submission` object from state, though.

I am sort of curious how this compares (or if it even works):

`Object.assign({}, ...Object.keys(state).filter(key => key === 'submission').map(key => state[key].data))`